drivers/platform/x86/thinkpad-wmi.c

```diff
@@ -477,10 +522,19 @@ static ssize_t store_setting(struct device *dev,
 	struct thinkpad_wmi *thinkpad = dev_get_drvdata(dev);
 	struct dev_ext_attribute *ea = to_ext_attr(attr);
 	int item_idx = (uintptr_t)ea->var;
-	const char *item = thinkpad->settings[item_idx];
+	char *item = thinkpad->settings[item_idx];
 	int ret;
 	size_t buffer_size;
 	char *buffer;
+	
+	/* Convert '\' to '/'. Please have a look at thinkpad_wmi_analyze. */
+	int spleng = 0;
+	int num = 0;
+	spleng = strlen(item);
+	for (num = 0; num < spleng; num++) {
```

Comment:
you can write it : `for (int i = 0; i < spleng; i++)`

Reply:
The module is currently built with -std=gnu89 (picked up  from kernel makefile) so trying to declare inside the for loop fails on compile.
error: 'for' loop initial declarations are only allowed in C99 or C11 mode.  Would prefer not to change the c standard used during compile for this module.

Reply:
ah, I though they had moved to C99 by now